### DISCUSSION TITLE
Coupling Configuration Release with ImageUpdateAutomation

### DISCUSSION THREAD

**Discussion by dieend** [fluxcd/flux2#1078] [Q&A] [answered]
Hello!
To describe problems that we have:
1. CI that build container image can take 20 minutes
2. Sometimes, a new deployment requires configuration changes that need to be tied with the new image version
3. If we are using ImageUpdateAutomation, we need to wait for the image to be ready, and then update the configuration so that the new Deployment rollout can use the new configuration and it doesn't impact old Deployment in case of disruption.

Ideally, when the image update happened, Only then it'll use the new configuration, and it's better if it happens in the same commit. I have no idea how it'll work considering that we only look at the docker registry and the docker registry doesn't have any metadata information outside of the label.

As an alternative, what we plan to do is doing a manual release.

Our chosen tagging system is alphabetical (instead of semver or numeric), and we can use `flux get image repository` to list the most recent deployable images.
What missing right now is setting a specific image version. From [here](https://github.com/fluxcd/flux2/discussions/936) I see we can use 
```
$ flux create image policy podinfo \
--image-ref=podinfo \
--select-semver=5.0.2
```
to set the image update for a specific semver version, but it seems the update will still happen remotely from the ImageUpdateAutomationController and no option for alphabetical tagging.

What I would like to happen is I can commit the selected image tag locally so I can couple it with configuration changes rather than it happened remotely outside of my control.

Is there a way to do this outside of manually updating the tag?

**Comment by squaremo** [comment, marked as answer]
It sounds like you are caught between the automation -- some of which is useful to you -- and needing to examine and possibly adapt updates that the automation makes. If I've got that right, then https://github.com/fluxcd/image-automation-controller/pull/121 may help. The automation would push to a branch, which you can then add commits to before merging.

**Reply by dieend** [reply]
Oh nice yes this can address the issue. Thanks!
